### PR TITLE
ListGroup doc #185

### PR DESCRIPTION
**Pull request by LiJinyao**
#185  Add ListGroup doc.

**API Change in ListGroupItem component:**
When I write doc, I found that using `color="success"` to set contextual class is not a good idea.

The `color` prop is no longer support, when you want to set a contextual class, just pass the name (one of “success”, “info”, “warning”, “danger”) to the component.

Since this component is not exists in doc yet and just released a few days, I think the change is acceptable.

### CODE REVIEW

**Comment by eddywashere** [comment]
@LiJinyao why remove `color` prop when that's a standard prop reactstrap uses for applying color classes? It's a good approach for a single choice based prop. We normally use boolean props to apply one off classes, classes that can be combined or used to apply a safe default.

**Comment by LiJinyao** [comment]
@eddywashere Oh I haven't notice that it is a standard prop. I'll change it back.
Why I make this change is `color="info"`  needs more type then just a `info` on off prop. But after your review I found that it is not convenient for generating color props.

File: docs/lib/Components/ListGroupPage.js [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+          <ListGroupCustomContentExample />
+        </div>
+        <pre>
+          <PrismCode className="language-jsx">
+            {ListGroupCustomContentExampleSource}
+          </PrismCode>
+        </pre>
+
+
+        <h4>Container Properties</h4>
```
Comment (eddywashere) [inline docs/lib/Components/ListGroupPage.js]: Can you remove the `container` docs from this

File: src/ListGroupItem.js [hunk 2]
```diff
@@ -33,6 +33,12 @@ const ListGroupItem = (props) => {
     'list-group-item'
   );
 
+  // Prevent click event when disabled.
+  if (disabled) {
```
Comment (eddywashere) [inline src/ListGroupItem.js]: I was going to ask to convert this to a normal component to make use of instance methods like [onClick in Button](https://github.com/reactstrap/reactstrap/blob/master/src/Button.js#L32-L41) but this is basically the same thing. Neat! I believe a new function is assigned each time it is rendered as disabled. Can you move this logic to a `const handleDisabledOnClick` that's assigned outside of the `ListGroupItem` function.

Comment (LiJinyao) [inline src/ListGroupItem.js]: OK，I'll do it as soon as possible :)

**Pull request merged**